### PR TITLE
XN-1387: add the functionality to delete a Hubspot contact

### PR DESCRIPTION
**Pull request by belie**
Adds ability to delete a hubspot in contact. 

This will be used by us when a user requests their prince account to be deleted

### CODE REVIEW

File: hubspot/client.go [hunk 1]
```diff
@@ -250,6 +250,37 @@ func (c *Client) ReadContact(email string, properties string) (*ContactOutput, E
 	return &contactOutput, ErrorResponse{}
 }
 
+// DeleteContact deletes a Contact in HubSpot
+func (c *Client) DeleteContact(contactID string) (bool, ErrorResponse) {
+
```
Comment (belie) [inline hubspot/client.go:255]: For the returns, I was somewhat torn between returning a bool, or just the ErrorResponse. Returning a bool seemed like an easier way for people to call this function to manage. However, I am not against just returning an ErrorResponse, either.

do any of you have any thoughts on this?

File: Makefile [hunk 2]
```diff
@@ -13,7 +13,7 @@ init:
 
 .PHONY: secrets-scan
 secrets-scan:
-	detect-secrets-hook --baseline ${DETECT_SECRETS_BASELINE} `git diff --cached --name-only)`
+	detect-secrets-hook --baseline ${DETECT_SECRETS_BASELINE} `git diff --cached --name-only`
```
Comment (karhidle) [inline Makefile:16]: nice catch

File: hubspot/client.go (hunk 1 above)
Comment (karhidle) [reply hubspot/client.go:255]: if this function doesn't return any content, then i think only returning the ErrorResponse should be enough. If ErrorResponse is nil, then it was successful.

File: hubspot/client_test.go [hunk 3]
```diff
@@ -489,14 +489,13 @@ func TestDeleteContact(t *testing.T) {
...
 
-			if hserr.Status != "" {
-				assert.Equal(t, http.StatusNotFound, hserr.StatusCode)
-				assert.Equal(t, false, is_deleted, "ensure the function did not delete ")
+			if tt.wantStatusCode == http.StatusNotFound {
+				assert.Equal(t, http.StatusNotFound, hserr.StatusCode, "")
 
 			} else {
-				assert.Equal(t, true, is_deleted, "ensure the function deleted ")
+				assert.Equal(t, "", hserr.Status, "ensure the function deleted ")
```
Comment (karhidle) [inline hubspot/client_test.go:498]: i thought on this case hserr should have been nil?

File: hubspot/hubspotiface/interface.go [hunk 4]
```diff
@@ -10,6 +10,7 @@ type HubSpotClient interface {
 	CreateContact(contactInput *hubspot.ContactInput) (*hubspot.ContactOutput, hubspot.ErrorResponse)
 	UpdateContact(contactID string, contactInput *hubspot.ContactInput) (*hubspot.ContactOutput, hubspot.ErrorResponse)
 	ReadContact(email string, properties string) (*hubspot.ContactOutput, hubspot.ErrorResponse)
+	DeleteContact(contactID string) (bool, hubspot.ErrorResponse)
```
Comment (cliffom) [inline hubspot/hubspotiface/interface.go:13]: Are you doing away with the `bool`?

File: hubspot/client_test.go (hunk 3 above)
Comment (belie) [reply hubspot/client_test.go:498]: It's returning an ErrorResponse like the other CRUD functions (as opposed to a reference to one). Think that should be changed for Delete?

File: hubspot/hubspotiface/interface.go (hunk 4 above)
Comment (belie) [reply hubspot/hubspotiface/interface.go:13]: thanks.

File: hubspot/client_test.go (hunk 3 above)
Comment (karhidle) [reply hubspot/client_test.go:498]: ah ok, yeah better to be consistent

**Review by karhidle — APPROVED** [review]

**Review by mpurdon — COMMENTED** [review]
lgtm

**Pull request merged**